### PR TITLE
RoR Budget App - Just sCENTse

### PR DESCRIPTION
**Pull request by 187jjay187**
## 🚀 Live Render Link <a name="live-demo"></a>

- [Live Render Link](https://rails-4prl.onrender.com)

## 🚀 Live Demo Video <a name="live-demo"></a>
- [Demo Video Link](https://www.loom.com/share/4781fedf3fc0412bac18c8ecac92df04)

## Challenge breakdown

Here is a suggestion of what you can do every day (just a suggestion, not mandatory):

### Day 1
- Set up the repository and tools.
- Create your models according to the ERD diagram. Don't forget about validations and tests.
- Implement the authentication and authorization. 

### Day 2

- Create the page to add a category.
- Create the home page.
- Create the page to add a transaction.
- Create the transactions page.

### Day 3

- Make sure your app is tested adequately.
- Deploy the project, and test for final details.
- Record a video for your project.
- Create a good README and PR description.

#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- Create unit and integration tests for all the most important components of your RoR application.

#### Technical requirements

- You should use Postgres as your database.
- You should use devise for authentication.
- You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
- You can use a view template engine of your choice (.erb, .slim, .haml).
- The project should be deployed and accessible online.
- Your database schema should reflect the following structure:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,160 @@
...
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://rails-4prl.onrender.com)
+
+- [Demo Video](https://www.loom.com/share/4dc400d2087a469ba4f38a773fdea9d7)
```
Comment (Lucas-Masaba) [inline README.md]: - [ ] That was an awesome presentation video. The just cents pun made cents :sweat_smile: . Unfortunately, according to the project requirements, the maximum length of the video should be 5 minutes. The video you share is 6 minutes and 15 seconds. Please adjust this to match the project requirements :pray: .

File: app/views/splash/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<div class="splash-container">
+  <div class="logo-container">
+    <%= image_tag('just_scentse_logo.png', alt: 'Example image', class: 'logo') %>
+    <p class="min-width: 1">Please create an account or log in</p>
+  </div>
+  <div class="sign-links">
+    <%= link_to 'Sign up', new_user_registration_path,  class: 'btn btn-primary sign-up' %> 
+    <%= link_to 'Log in'.html_safe, new_user_session_path, class: 'log-in' %>
```
Comment (Lucas-Masaba) [inline app/views/splash/index.html.erb]: - [x] The design guidelines for this project are to be followed as mentioned in the project requirements. The guidelines include 
 - Colours.
 - Typography: font face, size and weight.
 - Layout: composition and space between elements.
 Please make sure to use the write shade of blue provided to you in 
 the original design.
 ### Your shade of blue - #0d6efd
image
 ### Required shade of blue - #3778c2
image

File: app/views/splash/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="splash-container">
```
Comment (Lucas-Masaba) [inline app/views/splash/index.html.erb:1]: - [x] Your positioning of elements such as the sign-up and log-in buttons should be positioned similarly to that in the original design. In the original design, the logo is at the centre and the login and sign-in buttons are at the bottom. You can also see that the log in button has a background colour of  blue that stretches in length.
### Your Implementation
image
### Original design
image

File: app/views/devise/registrations/new.html.erb
Comment (Lucas-Masaba) [inline app/views/devise/registrations/new.html.erb:1]: - [x] Please make sure that this sign-up page has a similar layout of the positioning of the input fields and their labels as seen in the original design.
### Your Implementation
image
### Required 
image

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<h2 class="title">Log in</h2>
```
Comment (Lucas-Masaba) [inline app/views/devise/sessions/new.html.erb:1]: - [x] Please make sure that this log-in page has a similar layout of the positioning of the input fields and their labels as seen in the original design.
### Your Implementation
image
### Required
image

File: app/views/groups/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            <p class="date"><%= group.created_at.strftime('%d %B, %Y') %></p>
+            <p class="user">by <%= current_user.name %></p>
+          </div>
+        </div>
+      </div>
+    <% end %>
+  <% end %>
+
+  <div class="fixed-button">
+    <div class="text-center link-to">
```
Comment (Lucas-Masaba) [inline app/views/groups/index.html.erb]: - [x] Please make sure that this button appears at the bottom of the page like in the original design. The text in the button should have a white boundary as well.
### Your Implementation
image
### Required
image

File: app/views/groups/show.html.erb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+          <p class="amount">$<%= expense.amount %></p>
+        </div>
+        <p class="date"><%= expense.created_at.strftime('%d %B, %Y') %></p>
+      </div>
+    </div>
+  <% end %>
+
+  <div class="fixed-button">
+    <%= link_to new_group_expense_path(group_id: @group.id) do %>
+      <button class="btn btn-success but">Add new transaction</button>
```
Comment (Lucas-Masaba) [inline app/views/groups/show.html.erb:26]: - [x] Please make sure that this button appears at the bottom of the page like in the original design. The text in the button should have a white boundary as well.
### Your Implementation
image
### Required
image

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @187jjay187  👋 , 

Good job so far! 👍 
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights 🍾 
- Linter checks passing :heavy_check_mark: 
- Tests are passing :heavy_check_mark: 
- Met most requirements :+1: 

## Required Changes ♻️
- [ ] Most of the changes requested in the comments below are concerning the [design guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#design). You are required to use the colours, layout and typography provided to you by the [design guidelines](https://github.com/microverseinc/curriculum-rails/blob/main/capstone/rails_capstone.md#design).

## Optional suggestions 💭 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @187jjay187👋, 

### You did a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Nice UI design. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**